shared_workflow/platform_config.py

```diff
@@ -0,0 +1,144 @@
+    :param hostname: The name of the machine currently being run
+    :return: The platform being run, the path to the config for that platform
+    """
+    if hostname == "maui" or hostname == "mahuika":
+        hpc_platform = Platforms.NESI
+    elif hostname == "stampede2":
+        hpc_platform = Platforms.TACC
+    elif hostname == "nurion":
+        hpc_platform = Platforms.KISTI
+    elif hostname == "local":
+        hpc_platform = Platforms.LOCAL
+    else:
+        raise ValueError("Unexpected host given")
+
+    basename = f"platform_{hpc_platform.name.lower()}.json"
+
+    config_path = join(dirname(abspath(__file__)), "platform_configs", basename)
+    return hpc_platform, config_path
+
+
+platform, platform_config_path = determine_platform_config(host)
+platform_config = get_machine_config(config_path=platform_config_path)
+
+# Allows for loading data files from the templates and models directories
+for key, value in platform_config.items():
+    if isinstance(value, str) and "$workflow" in value:
+        platform_config[key] = value.replace("$workflow", WORKFLOW_DIR)
+
+errors = set(platform_config.keys()).symmetric_difference(
+    set([key.name for key in PLATFORM_CONFIG])
+)
+if errors:
+    missing_keys = []
+    extra_keys = []
+    for key in errors:
+        if key in platform_config:
+            extra_keys.append(key)
+        else:
+            missing_keys.append(key)
+    message = (
+        f"There were some errors with the platform config file {platform_config_path}."
+    )
+    if missing_keys:
+        message += f" Missing keys: {', '.join(missing_keys)}."
+    if extra_keys:
+        message += f" Additional keys found: {', '.join(extra_keys)}."
+    raise ValueError(message)
+
+# Dynamically generate the HPC enum
+
+HPC = Enum(
+    "HPC", platform_config[PLATFORM_CONFIG.AVAILABLE_MACHINES.name], module=__name__
+)
+
+
+def get_target_machine(process: Union[ProcessType, str, int]) -> HPC:
+    """
+    Takes in a process and returns the machine that task is expected to run on
+    :param process: The process to be checked. Represented by either the name index or ProcessType enum
+    :return: The HPC the task is to run on
+    """
+    if isinstance(process, str):
+        process = ProcessType[process]
+    elif isinstance(process, int):
+        process = ProcessType(process)
+    return HPC[platform_config[PLATFORM_CONFIG.MACHINE_TASKS.name][process.name]]
+
+
+def get_platform_specific_script(process: ProcessType, arguments: List[str]) -> str:
+    """
+    Returns the path to the script with arguments correctly formatted for the scheduler
+    :param process: The process to get the script for
+    :param arguments: Any arguments to be passed to the script
+    :return: The string representing the path to the script with the appropriate arguments to run it
+    """
+
+    # To prevent circular dependency
+    from scripts.schedulers.scheduler_factory import get_scheduler
```

Comment:
Hmmmm. We might need to think of a better long term solution

Reply:
Yeah, as annoying as it is I don't think there's a nicer way to split the platform config up. Making a wrapper for it in scheduler would be silly